hack/generated/pkg/reconcilers/deployment_error_classifier_test.go

```diff
@@ -43,47 +49,97 @@ func newError(inner ...armclient.DeploymentError) *armclient.DeploymentError {
 
 func Test_NilError_IsRetryable(t *testing.T) {
 	g := NewGomegaWithT(t)
-	g.Expect(reconcilers.ClassifyDeploymentError(nil)).To(Equal(reconcilers.DeploymentErrorRetryable))
+	expected := reconcilers.DeploymentErrorDetails{
+		Classification: reconcilers.DeploymentErrorRetryable,
+		Code:           reconcilers.UnknownErrorCode,
+		Message:        reconcilers.UnknownErrorMessage,
+	}
+	g.Expect(reconcilers.ClassifyDeploymentError(nil)).To(Equal(expected))
 }
```

Comment:
I think you could combine all these tests into a single test with subtests, using the `cases := struct{...}{...}` style we've used elsewhere. 

I can't decide whether you should, though.

Reply:
I thought about it when I initially wrote these. It ended up not seeming worth it though because the bulk of the lines of code in each test is defining the input error and the expected `DeploymentErrorDetails`, which is code that isn't duplicate. Combining into a single test with a loop to execute the list of inputs then would only really be saving us the duplicate assertion lines (and `NewGomegaWithT(t)`), which there is only one of in each test.

Given that the reduction in duplicate code isn't all that much, and the `struct {...}{...}` style tests have other disadvantages (harder to run just a single test, harder to break into them with a debugger) it felt better to go with this style. Especially because some of the cases do subtly differ from one another in a way that would increase the inner-loops complexity slightly in the `struct {...}{...}` testcase style (supporting multiple error inputs, supporting nil error input).

The `struct {...}{...}` feels more suited to tests where you have a moderate number of assertions which are being repeated across various styles of input. Bonus if you can craft that input itself in a loop somehow.

Happy to change this though if people disagree.

Reply:
Fair enough.